packages/openid4vc/src/openid4vc-verifier/OpenId4VcSiopVerifierService.ts

```diff
@@ -391,7 +393,29 @@ export class OpenId4VcSiopVerifierService {
       const suppliedSignature = await getSphereonSuppliedSignatureFromJwtIssuer(agentContext, requestSigner)
       builder.withSignature(suppliedSignature)
 
-      _clientId = suppliedSignature.did
+      if (requestSigner.method === 'x5c') {
+        let clientIdScheme: 'x509_san_dns' | 'x509_san_uri'
+
+        if (!requestSigner.issuer) {
+          throw new CredoError('Issuer must be provided to create x5c certificate protected authorization requests.')
+        }
+
+        const leafCertificate = X509Service.getLeafCertificate(requestSigner.chain)
+        if (leafCertificate.sanDnsNames?.includes(requestSigner.issuer)) {
+          clientIdScheme = 'x509_san_dns'
+        } else if (leafCertificate.sanUriNames?.includes(requestSigner.issuer)) {
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          clientIdScheme = 'x509_san_uri'
+        } else {
+          throw new CredoError('The x5c issuer must be in the SAN-DNS or SAN-URI names of the leaf certificate.')
+        }
+
+        // TODO:
+        // builder.withClientIdScheme(clientIdScheme)
```

Comment:
still needs to be supported in sphereons libs?

Reply:
@auer-martin

Reply:
yes needs to be implemented